### PR TITLE
Disable windows hardware timestamps

### LINKED ISSUE #1614: Disable hardware timestamps on Windows
We have seen too many recordings on Windows whose timestamps are out of sync. Currently, we use the hardware timestamps given by the camera. In the cases of the out-of-sync recordings, these timestamps either had a constant offset (multiple minutes) or drifted during the recording. We will be disabling the hardware timestamp feature and reverting to software timestamps on Windows.

Software frame timestamps are measured by the host computer when the frame has been received from the camera instead of when the frame started exposing (hardware timestamp). This means that software timestamps carry an inherent delay of a few milliseconds.

In the current situation, we are not able to continue supporting hardware timestamps since they do not provide the required robustness on Windows. If you require hardware timestamps you will have to use Pupil Capture on Linux or macOS.

### PR DESCRIPTION
**Pull request by pfaion**
Resolves #1614

### CODE REVIEW

File: pupil_src/shared_modules/video_capture/uvc_backend.py [hunk 1]
```diff
@@ -180,7 +182,21 @@ def verify_drivers(self):
...
-            self.ts_offset = 0.0
+            if platform.system() == "Windows":
+                # NOTE: Hardware timestamps seem to be broken on windows. Needs further
+                # investigation! Disabling for now.
+                # TODO: Find accurate offsets for different resolutions!
+                offsets = {"ID0": -0.015, "ID1": -0.015, "ID2": -0.07}
+                match = re.match(r"Pupil Cam\d (?P<cam_id>ID[0-2])", self.name)
+                if not match:
+                    logger.debug(f"Could not parse camera name: {self.name}")
+                    self.ts_offset = 0.0
```
Comment (papr) [inline pupil_src/shared_modules/video_capture/uvc_backend.py]: Maybe implement a default offset of  `-0.01`? Also apply below for unknown cams
```suggestion
                    self.ts_offset = -0.01
```

File: pupil_src/shared_modules/video_capture/uvc_backend.py [hunk 2]
```diff
@@ -180,7 +182,21 @@ def verify_drivers(self):
...
+                else:
+                    self.ts_offset = offsets[match.group("cam_id")]
+
+            else:
+                # use software timestamps
```
Comment (papr) [inline pupil_src/shared_modules/video_capture/uvc_backend.py]:
```suggestion
                # use hardware timestamps
```

File: pupil_src/shared_modules/video_capture/uvc_backend.py [hunk 3]
```diff
@@ -180,7 +182,21 @@ def verify_drivers(self):
...
+                self.ts_offset = 0.0
```
Comment (papr) [inline pupil_src/shared_modules/video_capture/uvc_backend.py]:
```suggestion
                self.ts_offset = None
```

File: pupil_src/shared_modules/video_capture/uvc_backend.py (hunk 1 above)
Comment (pfaion) [reply pupil_src/shared_modules/video_capture/uvc_backend.py]: @papr Hm, shouldn't we maybe use software timestamps for unknown cameras? (This is what this was supposed to be)

Comment (pfaion) [reply pupil_src/shared_modules/video_capture/uvc_backend.py]: In this case I'd do:
```suggestion
                    self.ts_offset = None
```

Comment (papr) [reply pupil_src/shared_modules/video_capture/uvc_backend.py]: `self.ts_offset = None` triggers hardware timestamps. In this specific case, we are on Windows. We do not want to use hardware timestamps in any case on Windows.

**Comment by pfaion** [comment]
@papr :shipit:

**Pull request merged**